File: sorting/selection_sort.py
data=[16,25,39,27,12,8,45,63]

def selectionSort(array):
    for i in range(len(array)-1):
        #設未排序陣列中的第一個元素為最小的
        min_index=i
        for j in range(i+1,len(array)):
            #找到未排序中最小元素的index
            if array[min_index]>array[j]:
                min_index=j
        #將最小的數字放到已排序數列中的最後面
        array[i],array[min_index]=array[min_index],array[i]
    return array

print(selectionSort(data))

File: sorting/selectionsort.c
#include <stdio.h>
#define SIZE 8

void selectionSort(int *);
void printArray(int *);

int main(){
	int data[SIZE]={16,25,39,27,12,8,45,63};
	selectionSort(data);
	printArray(data);
	return 0;
}

void selectionSort(int *array){
	void swap(int *,int *);
	int i,j,min_index;
	for(i=0;i<SIZE-1;i++){
		//initiaize the index of the unsorted array
		min_index=i;
		
		//find the smallest nuber in the unsorted array
		for(j=i+1;j<SIZE;j++){
			if(array[min_index]>array[j]){
				min_index=j;
			}
		}
		
		//put the smallest number into the last of the sorted array
		swap(&array[min_index],&array[i]);
	}
}

void swap(int *a, int *b){
	int temp;
	temp=*a;
	*a=*b;
	*b=temp;
}

void printArray(int *array){
	int i;
	for(i=0;i<SIZE;i++){
		printf("%d ",array[i]);
	}
}
